hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/CryptoDeleteAllowanceHandler.java

```diff
@@ -152,10 +153,18 @@ private void deleteNftSerials(
 
                 validateTrue(nft != null, INVALID_NFT_ID);
                 // Check if owner owns the nft
-                validateTrue(isValidOwner(nft, owner.accountNumber(), token), SENDER_DOES_NOT_OWN_NFT_SERIAL_NO);
+                // FUTURE: owner will be updated to use an AccountID instead of a long account number
+                // Issue #7243
+                AccountID accountOwner = AccountID.newBuilder()
```

Comment:
We already have owner (Account) from line 149, couldn't we use the AccountID from that?
Also, if we cannot use the owner, we should get the shard and realm from the tokenId on the assumption the shard/realm of the account matches the token it owns.

Reply:
The plan is to use the owner Account from line 149. Currently Account does not have a getAccountId() method, only a getAccountNum(). Issue #7243 will correct that. The code here is very temporary until issue #7243 can be implemented, probably this week.